builtin/credential/ldap/path_config_rotate_root.go

```diff
@@ -0,0 +1,99 @@
+package ldap
+
+import (
+	"context"
+
+	"github.com/go-ldap/ldap/v3"
+
+	"github.com/hashicorp/vault/sdk/helper/base62"
+	"github.com/hashicorp/vault/sdk/helper/ldaputil"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+func pathConfigRotateRoot(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "config/rotate-root",
+
+		DisplayAttrs: &framework.DisplayAttributes{
+			OperationPrefix: operationPrefixLDAP,
+			OperationVerb:   "rotate",
+			OperationSuffix: "root-credentials",
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathConfigRotateRootUpdate,
```

Comment:
If we know we'll write to Vault's storage in the handler, we can forward this request with `ForwardPerformanceStandby/Secondary`. That'll avoid us rotating the LDAP password, getting a read-only storage error from Vault, and going through this handler again (on the active) once it gets automatically forwarded.

Reply:
Here's[ an example](https://github.com/hashicorp/vault-plugin-auth-azure/blob/main/path_rotate_root.go#L29-L30) on how this can be achieved from one of our repos in case you were looking for one! :)